src/webgpu/shader/validation/expression/call/builtin/const_override_validation.ts

```diff
@@ -312,3 +316,74 @@ export function unique<T>(...arrays: Array<readonly T[]>): T[] {
   }
   return [...set];
 }
+
+interface FloatLimits {
+  positive: {
+    max: number;
+  },
+  negative: {
+    min: number;
+  },
+  emax: number;
+}
+
+/**
+ * Provides an easy way to validate steps in an equation that will trigger a validation error with
+ * constant or override values due to overflow/underflow. Typical call pattern is:
+ * 
+ * const vCheck = new ConstantOrOverrideValueChecker(t, Type.f32);
+ * const c = vCheck.checkedResult(a + b);
+ * const d = vCheck.checkedResult(c * c);
+ * const expectedResult = vCheck.allChecksPassed();
+ */
+export class ConstantOrOverrideValueChecker {
+  #allChecksPassed = true;
+  #floatLimits?: FloatLimits;
+  #quantizeFn: QuantizeFunc<number>;
+
+  constructor(private t: ShaderValidationTest, type: ScalarType) {
```

Comment:
I don't know JS well enough but I presume this makes class fields `t` and `type`. Otherwise you couldn't access `t` to skip the test later.

Reply:
Yeah, it makes them private fields. I had to look this up along with the # syntax, since I don't normally use them

Reply:
Yep, though it's a typescript feature, not a JavaScript one.

Happy to remove any syntax that feels out of place or confusing, I'm just used to using these in my own code.

Reply:
I am fine with leaving it as is, I don't find it particularly cryptic, was just a syntax I wasn't familar with.

Reply:
TypeScript allows for private attributes to also be declared as:

`private someValue: number;`

If that reads better.

Reply:
I am fine with either way